class.jetpack-gutenberg.php

```diff
@@ -801,4 +815,46 @@ public static function get_extensions_preset_for_variation( $preset_extensions_m
 
 		return $preset_extensions;
 	}
+
+
+	/**
+	 * Set the availability of the block as the editor
+	 * is loaded.
+	 *
+	 * @param string $slug Slug of the block.
+	 */
+	public static function set_availability_for_plan( $slug ) {
+		$is_available = true;
+		$plan         = '';
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			if ( ! class_exists( 'Store_Product_List' ) ) {
+				require WP_CONTENT_DIR . '/admin-plugins/wpcom-billing/store-product-list.php';
+			}
+			$features_data = Store_Product_List::get_site_specific_features_data();
+			$is_available  = in_array( $slug, $features_data['active'], true );
+			if ( isset( $features_data['available'][ $slug ] ) && ! empty( $features_data['available'][ $slug ] ) ) {
+				$plan = $features_data['available'][ $slug ][0];
+			}
+		} elseif ( ! jetpack_is_atomic_site() ) {
+			/*
+			 * If it's Atomic then assume all features are available
+			 * otherwise check against the Jetpack plan.
+			 */
+			$is_available = Jetpack_Plan::supports( $slug );
+			$plan         = Jetpack_Plan::get_minimum_plan_for_feature( $slug );
+		}
+		if ( $is_available ) {
+			self::set_extension_available( $slug );
+		} else {
+			self::set_extension_unavailable(
+				$slug,
+				'missing_plan',
+				array(
+					'required_feature' => self::remove_extension_prefix( $slug ),
```

Comment:
Since we're passing in the block name without the `jetpack/` do we need this?

Reply:
Yes, I think I was doing it more as a safeguard. You're right, it's not needed!